### ISSUE TITLE
Wire plug in Connection1.jpg Wrong

### ISSUE THREAD

**Issue by brosskgm** [markniu/Bed_Distance_sensor#204]
Just received your unit Today. Installing on Mega X with SKR V1.3 TMC 2208 and Marlin 2.1.2.4

**Comment by markniu** [comment]
right, the SCL_PIN means the CLK pin.
there should some version string after M102 S-1 , else it's working.
and I just checked this skr v1.3 , there are signal filter capacitors on all the endstop pins, so we can not use those pins
![image](https://github.com/user-attachments/assets/eada001e-47fa-4c5f-b3f2-c2c996fdb230)

please choose other pins like the EXP1 EXP2 or TFT

**Comment by brosskgm** [comment]
Hi Mark, Thanks for the quick reply.

Do you have any instructions for the EXP1 or EXP2. Already using the TFT pins for the screen.

Thanks
Bob

**Comment by markniu** [comment]
are there some message returned after M102 S-1  like `pandapi3d.com v1.2c`

**Comment by brosskgm** [comment]
M102 S-1
Returns OK

**Comment by brosskgm** [comment]
Hi Mark,

I decided to get the SKR V1.4 board since it has the Survo and Probe pins seperate. Will get in touch with you then.

Thanks
Bob

**Comment by brosskgm** [comment]
Hi Mark,

Got the SKR V1.4 all installed. Now when I send the M102 S-1 Command it gives unknown command.

**Comment by markniu** [comment]
I just check that the gcode M102 S-1 and some features are not working in the  marlin mainline, could you use this https://github.com/markniu/MarlinPULL  that I have just updated to the latest marlin.

**Comment by brosskgm** [comment]
Went through and changed the Config and Pins files and now when I try to compile it gives me the attached error that it's not a Valid Marlin 2.x project.

![VisualError](https://github.com/user-attachments/assets/eaf2d2a2-00e8-4685-a859-571b3e164964)

**Comment by markniu** [comment]
I just checked there is no problem while compiling.
![image](https://github.com/user-attachments/assets/568c9df0-b449-4a7d-8807-fae0023a7a4c)


about your error, maybe you can check your path or the folder name, https://github.com/MarlinFirmware/Marlin/issues/23431#issuecomment-1003749157



> Went through and changed the Config and Pins files and now when I try to compile it gives me the attached error.
>

**Comment by brosskgm** [comment]
I'll check. This was on the SKR V1.4 - It compikles fine with all the other Marlin directories. Even when I select to open directory and select it, it said the same erro. I downloaded 2.1.2.5 bug fix and it didn't give the error. I'll see if I can make the config changes in the bug fix and see if it starts to work.

**Comment by brosskgm** [comment]
I had to take all the dashes(-) and dots out of the folder name it it accepted it.

The compile now didn't like it. 89 errors.

![CompileError-4](https://github.com/user-attachments/assets/75d9c68f-524f-4229-a010-469113361fe5)
![CompileError-3](https://github.com/user-attachments/assets/3f67be0e-74f9-4e5c-bb74-0a1286aaac1e)
![CompileError-2](https://github.com/user-attachments/assets/996b3245-17f7-4c22-8108-97bb6c5b632d)
![CompileError-1](https://github.com/user-attachments/assets/0a853c08-9f4e-4b0f-9875-f34318b9f52f)

**Comment by brosskgm** [comment]
After clearing the problems/errors it now only fails the build with the same error as above.

Marlin/src/module/probe.cpp: In static member function 'static float Probe::probe_at_point(const_float_t, const_float_t, ProbePtRaise, uint8_t, bool, bool, const_float_t, const_float_t, bool)':
Marlin/src/module/probe.cpp:995:43: error: 'class BDS_Leveling' has no member named 'read'; did you mean 'bd_read'?
  995 |     measured_z = current_position.z - bdl.read(); // Difference between Z-home-relative Z and sensor reading
      |                                           ^~~~
      |                                           bd_read
Compiling .pio/build/LPC1768/src/src/module/stepper/indirection.cpp.o
*** [.pio/build/LPC1768/src/src/module/probe.cpp.o] Error 1

**Comment by markniu** [comment]
yes, please change the` bdl.read();` to `bdl.bd_read();` in the probe.cpp,  that I have updated to the [pull](https://github.com/MarlinFirmware/Marlin/pull/27243/files#diff-9252e5d1c9a93c8220c9a7cae4a8dfec5a45dce3f36afe20c66063278ef79954) too

> After clearing the problems/errors it now only fails the build with the same error as above. Here is the line in probe.cpp it's talking about and photo from the file.
>

**Comment by brosskgm** [comment]
It compiled. Will run it on the printer tomorrow. Will let you know how it does.

**Comment by brosskgm** [comment]
It all worked. Looks like I'll need to work on the sensor heigh.

![BD-](https://github.com/user-attachments/assets/27e3a32a-6f58-4870-ae73-f3cc6ed67e3b)
![BD-2](https://github.com/user-attachments/assets/dc435587-f02c-49b4-95ae-32b58e7b4688)
![BD-3](https://github.com/user-attachments/assets/db4fa473-7919-4f68-978f-3a06f72111b5)
![BD-4](https://github.com/user-attachments/assets/27d1886e-5afc-47a5-a32d-093067ba6665)

It doesn't touch the bed, but looks like I will need to find a way to raise it.

![BD-5](https://github.com/user-attachments/assets/54e3ca00-bf2d-49f9-a148-ce4feeb1ab5f)

**Comment by brosskgm** [comment]
Or maby lower it. The Mic say little over 4mm height.

**Comment by markniu** [comment]
right, the first calibration data is too high, We can simply calculate the sensor to the bed is about 629*0.005 = 3.14 mm ,

**Comment by brosskgm** [comment]
The mount is the longest one on the three in the stl files.

**Comment by brosskgm** [comment]
I also printed the BL-Touch mount. Might have to see if I can find a couple with different heigths.

**Comment by brosskgm** [comment]
Yep. Found one 3mm shorter. Will print it up tomorrow and re-do it all.

**Comment by brosskgm** [comment]
Printed a new BLTouch mount. Extended the height 1.5mm and after calibration M102 S-5 if doesn't give the Bad Sensor height.

Maybe since I have the probe in the front of nozzle, might not really need to worry if I have it go to the regular home location 0 as before probe was installed.

![NewSensorHeight](https://github.com/user-attachments/assets/ca043e56-55b5-414e-a14d-ec5943436514)

Getting closer.

**Comment by markniu** [comment]
From the raw calibrate data, it seems that you need to check the steps of the Z axis or the movement while calibrating. 
since if the first point is 149 then second data should be at least about 149  + 20 = 169. 
the z axis moves at 0.1mm each time while calibration, but seems move 0.05mm.

**Comment by brosskgm** [comment]
Not sure where to find that one. The Z_STEPS in configuration.h are only for ABL or MBL and not enabled.

**Comment by brosskgm** [comment]
Could only find AXIS steps.

![Z_STEPS](https://github.com/user-attachments/assets/c989a42e-ac0d-4da8-9259-a9562c9e1092)

**Comment by brosskgm** [comment]
Here is what ahppens when G28 is sent. It keeps re-homing again over and over. The only way to stop it is turn off the printer.

**Comment by brosskgm** [comment]
Here is what ahppens when G28 is sent. It keeps re-homing again over and over. The only way to stop it is turn off the printer. Never get the chance to send G1 Z0

![Image](https://github.com/user-attachments/assets/82b4bca6-0608-4071-b5d6-262721b39840)

**Comment by brosskgm** [comment]
Funny they say I closed this yesterday and most was deleted. I didn't log in yesterday at all. I'll keep trying to figure it out.
